api/src/shared/application/scripts/parsers.js

```diff
@@ -0,0 +1,52 @@
+import Joi from 'joi';
+
+import { checkCsvHeader, parseCsvWithHeader } from '../../infrastructure/helpers/csv.js';
+
+/**
+ * Create a parser for a CSV file with the given columns schema
+ * @param {Record<string, Joi.Schema>} columnsSchema
+ * @returns {Promise<Array<Record<string, any>>>} parsed data
+ */
+export function csvFileParser(columnsSchema = []) {
+  return async (filePath) => {
+    const columnNames = columnsSchema.map((column) => column.name);
+
+    await checkCsvHeader({ filePath, requiredFieldNames: columnNames });
+
+    return parseCsvWithHeader(filePath, {
+      header: true,
+      skipEmptyLines: true,
+      transform: (value, columnName) => {
+        const column = columnsSchema.find((column) => column.name === columnName);
```

Comment:
Nommage correct en anglais : 

```suggestion
/**
 * Create a parser for a CSV file with the given column schema
 * @param {Record<string, Joi.Schema>} columnSchema
 * @returns {Promise<Array<Record<string, any>>>} parsed data
 */
export function csvFileParser(columnSchema = []) {
  return async (filePath) => {
    const columnNames = columnSchema.map((column) => column.name);

    await checkCsvHeader({ filePath, requiredFieldNames: columnNames });

    return parseCsvWithHeader(filePath, {
      header: true,
      skipEmptyLines: true,
      transform: (value, columnName) => {
        const column = columnSchema.find((column) => column.name === columnName);
```